src/SymmetricPositiveDefinite.jl

```diff
@@ -0,0 +1,482 @@
+using LinearAlgebra: diagm, diag, eigen, eigvals, eigvecs, Symmetric, Diagonal, factorize, tr, cholesky, LowerTriangular
+
+@doc doc"""
+    SymmetricPositiveDefinite{N} <: Manifold
+
+The manifold of symmetric positive definite matrices, i.e.
+
+```math
+\mathcal P(n) =
+\bigl\{
+    x \in \mathbb R^{n\\times n} :
+    \xi^\mathrm{T}x\xi > 0 \text{ for all } \xi \in \mathbb R^{n}\backslash\{0\}
+\bigr\}
+```
+
+# Constructor
+
+    SymmetricPositiveDefinite(n)
+
+generates the manifold $\mathcal P(n) \subset \mathbb R^{n\times n}$
+"""
+struct SymmetricPositiveDefinite{N} <: Manifold end
+SymmetricPositiveDefinite(n::Int) = SymmetricPositiveDefinite{n}()
+
+@doc doc"""
+    manifold_dimension(::SymmetricPositiveDefinite{N})
+
+returns the dimension of the manifold [`SymmetricPositiveDefinite`](@ref) $\mathcal P(n), N\in \mathbb N$, i.e.
+```math
+    \frac{n(n+1)}{2}    
+```
+"""
+@generated manifold_dimension(::SymmetricPositiveDefinite{N}) where {N} = div(N*(N+1), 2)
+
+@doc doc"""
+    LinearAffineMetric <: Metric
+
+The linear affine metric is the metric for symmetric positive definite matrices, that employs
+matrix logarithms and exponentials, which yields a linear and affine metric.
+"""
+struct LinearAffineMetric <: RiemannianMetric end
+@traitimpl HasMetric{SymmetricPositiveDefinite,LinearAffineMetric}
+# Make this metric default, i.e. automatically convert
+convert(::Type{SymmetricPositiveDefinite{N}}, M::MetricManifold{SymmetricPositiveDefinite{N},LinearAffineMetric}) where N = M.manifold
+convert(::Type{SymmetricPositiveDefinite}, M::MetricManifold{SymmetricPositiveDefinite{N},LinearAffineMetric}) where N = M.manifold
+convert(::Type{MetricManifold{SymmetricPositiveDefinite{N},LinearAffineMetric}}, M::SymmetricPositiveDefinite{N}) where N = MetricManifold(M, LinearAffineMetric())
+
+@doc doc"""
+    LogEuclideanMetric <: Metric
+
+The LogEuclidean Metric consists of the Euclidean metric applied to all elements after mapping them
+into the Lie Algebra, i.e. performing a matrix logarithm beforehand.
+"""
+struct LogEuclideanMetric <: RiemannianMetric end
+
+@doc doc"""
+    LogCholeskyMetric <: Metric
+
+The Log-Cholesky metric imposes a metric based on the Cholesky decomposition as
+introduced by
+> Lin, Zenhua: "Riemannian Geometry of Symmetric Positive Definite Matrices via
+> Cholesky Decomposition", arXiv: [1908.09326](https://arxiv.org/abs/1908.09326).
+"""
+struct LogCholeskyMetric <: RiemannianMetric end
+
+cholesky_to_spd(l,w) = (l*l', w*l' + l*w')
+tangent_cholesky_to_tangent_spd!(l,w) = (w .= w*l' + l*w')
+spd_to_cholesky(x,v) = spd_to_cholesky(x,cholesky(x).L,v)
+function spd_to_cholesky(x,l,v)
+    a = l\v
+    w = transpose(l\(a'))
```

Comment:
Classically I wanted to compute `inv(l)*v*inv(transpose(l))` – and `l` is already a cholesky factor – so – what do you mean by computing a decomposition when `l` is already cholesky?

Reply:
Oh, sorry, I missed that... it's just one line above.

Reply:
Ok, I'll use that then – the complicated way to `a` was mainly because I get very much pointed to never use inv (in Matlab) and that still was in my head ;) Thanks again for the remarks.